src/classes/Advancement_Adapter.cls

```diff
@@ -0,0 +1,101 @@
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2017
+* @group TDTM
+* @group-content ../../ApexDocContent/TDTM.htm
+* @description Provides an adapter for other packages to be able to inject customizations into HEDA.
+**/
+
+global virtual with sharing class Advancement_Adapter {
+
+    /**
+     * @description To be called by the HEDA TDTM trigger dispatcher. It returns the HEDA TDTM tokens with the new one from Adv.
+     *
+     * @return A list of HEDA tokens.
+     */
+    global virtual List<TDTM_Global_API.TdtmToken> getDefaultTdtmConfigTokens() {
+        if(!Advancement_Info.useAdv()) {
+            throw new AdapterException('Invalid call. This function is reserved for Salesforce use.');
+        }
+
+        return TDTM_Global_API.getDefaultTdtmConfig();
+    }
+
+    /**
+    *  @description This method is used to inject the NPSP rollup logic into HEDA Trigger Handler.
+    */
+    global virtual void rollupAccount(List<Id> listAccountId) {
+        if(!Advancement_Info.useAdv()) {
+            throw new AdapterException('Invalid call. This function is reserved for Salesforce use.');
+        }
+
+        //No logic for HEDA
+    }
+
+    /**
+     * @description This method is used to inject the NPSP opportunity logic info HEDA Trigger Handler.
+     */
+    global virtual void moveOppsToContactAccount(Map<Id,Id> mapContactIdAccountIdNew, Map<Id,Id> mapContactIdAccountIdOld) {
+        if(!Advancement_Info.useAdv()) {
+            throw new AdapterException('Invalid call. This function is reserved for Salesforce use.');
+        }
+
+        //No logic for HEDA
+    }
+
+    /**
+     * @description This method is used to populate NPSP's primary contact field in HEDA Trigger Handler
+     */
+    global virtual void primaryContact(Account a, Id contactId) {
+        if(!Advancement_Info.useAdv()) {
+            throw new AdapterException('Invalid call. This function is reserved for Salesforce use.');
+        }
+
+        //No Logic for HEDA
+    }
+
+    /**
+     * @description This method is used to populate NPSP's system account field in HEDA Trigger Handler
+     */
+    global virtual void configAccountForContact(Account a) {
+        if(!Advancement_Info.useAdv()) {
+            throw new AdapterException('Invalid call. This function is reserved for Salesforce use.');
+        }
+
+        //No Logic for HEDA
+    }
+
+    /**
+     * @description This is the expection class for Advancement Adapter class
+     */
+    global class AdapterException extends Exception{}
```

Comment:
Do we need an Advancement_Adapter_TEST class to exercise these methods that throw exceptions?

Reply:
Yes, I will add the unit test to verify it. Also, I think we need to skip the test if Adv is installed.